### DISCUSSION TITLE
[Permissions] Réfléchir à un nouveau système de permissions plus fonctionnel

### DISCUSSION THREAD

**Discussion by bruhnild** [GeotrekCE/Geotrek-admin#3904] [Ideas]
Le système de permissions actuel est trop complexe.
Les utilisateurs sont souvent confus sur ce qu'ils doivent autoriser ou non en fonction des cas d'usage et des utilisateurs.

**Comment by camillemonchicourt** [comment]
Ah ouais ? A quel niveau ? Peux-tu préciser ?
J'ai jamais trop eu ce genre de retours.
L'interface de gestion des permissions est vraiment pas lisible et clair.
Mais le système et l'organisation des permissions est plutôt bien adapté et fonctionnel.
Car il est précis et permet ainsi de gérer finement les permissions objet par objet.
Et l'ajout du système par structure permet un garde fou intéressant et correspondant à la majorité des cas d'usage où on veut qu'un utilisateur puisse modifier les objets de sa structure, mais pas ceux des autres structures.

**Comment by babastienne** [comment]
Personnellement ce n'est pas le système des structure qui m'embête mais le fait qu'il y a beaucoup trop de droits différents.

Il faudrait faire des permissions "fonctionnelle", par thématique. Un peu l'équivalent de groupes finalement mais avec la possibilité d'aller plus finement dans le détail si besoin.

Pouvoir facilement pour un utilisateur être capable de dire : lui peux accéder au module signalétique pour lire / écrire / supprimer, suffit dans la plupart des cas. Devoir pour chaque typologie d'objet (les lames, les signalétiques, les type des signalétiques, les type de lames, etc.) régler ces paramètres devient laborieux et contre-intuitif. Je ne dis pas qu'il ne faut pas donner cette possibilité, mais juste que ça devrait être secondaire et que la gestion des modules dans leur globalité ne doit pas être noyée au milieu de toutes ces infos secondaires.

**Reply by babastienne** [reply]
Au final on a pas mal de territoires qui ne comprennent pas vraiment la gestion des droits et on se retrouve à faire la paramétrage pour eux.

**Comment by camillemonchicourt** [comment]
Oui par contre, ce qui est peu lisible c'est qu'on mélange les permissions de bases dans les modules et les permissions sur la gestion des listes de valeurs dans l'AdminSite.
C'est utile de pouvoir gérer cette finesse, mais c'est peu compréhensible et lisible actuellement.

C'est pour ça que je disais que je trouve le système de permissions bien fin, pertinent et adapté aux besoins, mais que c'est l'interface de gestion de ceux-ci qui est pas top, selon moi.

**Reply by babastienne** [reply]
Ah oui bah dans ce cas je suis d'accord en effet. ce qu'il faut revoir c'est l'interface et probablement des groupements de permissions plus génériques / métiers.
